galaxychop/galaxychop.py

```diff
@@ -65,52 +81,94 @@ class Galaxy:
     ---------
     """
 
-    x_s = attr.ib()
-    y_s = attr.ib()
-    z_s = attr.ib()
-    vx_s = attr.ib()
-    vy_s = attr.ib()
-    vz_s = attr.ib()
-    m_s = attr.ib()
-
-    x_dm = attr.ib()
-    y_dm = attr.ib()
-    z_dm = attr.ib()
-    vx_dm = attr.ib()
-    vy_dm = attr.ib()
-    vz_dm = attr.ib()
-    m_dm = attr.ib()
-
-    x_g = attr.ib()
-    y_g = attr.ib()
-    z_g = attr.ib()
-    vx_g = attr.ib()
-    vy_g = attr.ib()
-    vz_g = attr.ib()
-    m_g = attr.ib()
-
-    E_tot_dark = attr.ib(default=None)
-    E_tot_star = attr.ib(default=None)
-    E_tot_gas = attr.ib(default=None)
+    x_s = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+    y_s = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+    z_s = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+    vx_s = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+    vy_s = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+    vz_s = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+    m_s = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+
+    x_dm = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+    y_dm = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+    z_dm = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+    vx_dm = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+    vy_dm = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+    vz_dm = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+    m_dm = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+
+    x_g = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+    y_g = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+    z_g = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+    vx_g = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+    vy_g = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+    vz_g = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+    m_g = attr.ib(validator=attr.validators.instance_of(u.Quantity))
+
+    eps_s = attr.ib(default=0. * u.kpc,
+                    validator=attr.validators.instance_of(u.Quantity))
+    eps_dm = attr.ib(default=0. * u.kpc,
+                     validator=attr.validators.instance_of(u.Quantity))
+    eps_g = attr.ib(default=0. * u.kpc,
+                    validator=attr.validators.instance_of(u.Quantity))
+
+    Etot_dm = attr.ib(default=None)
+    Etot_s = attr.ib(default=None)
+    Etot_g = attr.ib(default=None)
 
     components_s = attr.ib(default=None)
     components_g = attr.ib(default=None)
     metadata = attr.ib(default=None)
 
-    def energy(self, eps=0):
+    def __change_units_to_array__(f):
+        def new_method(self, *args, **kwargs):
```

Comment:
Should the name of the method refer to what it does?

Reply:
I don't really know. I only chose the name to know quickly what the decorator does.